src/core/mac/mac.cpp

```diff
@@ -1819,6 +1851,43 @@ void Mac::ResetCounters(void)
     memset(&mCounters, 0, sizeof(mCounters));
 }
 
+#if OPENTHREAD_CONFIG_MAC_RETRY_SUCCESS_HISTOGRAM_ENABLE
+const uint32_t *Mac::DirectRetrySuccessHistogramGet(uint8_t *aNumberOfEntries)
+{
+    if (mMaxFrameRetriesDirect > kDefaultMaxFrameRetriesDirect)
```

Comment:
I believe it should be minimum of OPENTHREAD_CONFIG_MAC_RETRY_SUCCESS_HISTOGRAM_MAX_SIZE_COUNT_DIRECT and mMaxFrameRetriesDirect. Remember that first define is set to default number of retransmissions only by default, but user can change it.